samples/csx/HttpSyncStart/run.csx

```diff
@@ -0,0 +1,24 @@
+#r "Microsoft.Azure.WebJobs.Extensions.DurableTask"
+#r "Newtonsoft.Json"
+
+using System.Net;
+using System.Net.Http.Headers;
+
+public static async Task<HttpResponseMessage> Run(
+    HttpRequestMessage req,
+    DurableOrchestrationClient starter,
+    string functionName,
+    TraceWriter log)
+{
+    // Function input comes from the request content.
+    dynamic eventData = await req.Content.ReadAsAsync<object>();
+    string instanceId = await starter.StartNewAsync(functionName, eventData);
+    
+    log.Info($"Started orchestration with ID = '{instanceId}'.");
+    
+    return await starter.CreateCheckStatusResponse(
+                req,
+                instanceId,
+                TimeSpan.FromSeconds(2),
```

Comment:
Let's not hardcode the timeout and retry intervals. Instead, let's make them optional query string parameters. That will make this start function more generic. For example, I would imagine someone could do the following:

```http
POST /orchestrators/myfunction/wait?timeout=10&retryInterval=1
```

The `timeout` and `retryInterval` parameters will be interpreted in seconds. If they are not specified, then we can have default values. If we want to support sub-second values, then we could allow them to use floating point numbers, like `retryInterval=0.5`.

For the default values, I would suggest making the timeout at least 10 seconds and the retry interval at least 1 second. Retrying too rapidly could create a lot of I/O overhead on the storage account. I also don't expect that orchestrations will normally complete that quickly (though I could be wrong - feel free to disagree with this one).

Reply:
Actually I was thinking the same yesterday. It will be great if we can define the timeout details with every call instead being driven by the server side. Great change!